src/ui/SubredditPage.tsx

```diff
@@ -19,27 +19,42 @@ export default class SubredditPage extends Page implements common.SubredditView
   private _items: RedditPost[] = [];
   private _mode: ViewMode;
   private loading: boolean;
-  @getById private collectionView: CollectionView;
+  @getById private listView: ListView<RedditPost>;
 
   constructor(properties?: Properties<SubredditPage>) {
     super(properties);
     this.append(
-      <CollectionView id='collectionView'
+      <ListView
+          id='listView'
           stretch
           background='#f5f5f5'
-          cellType={() => this._mode}
-          cellHeight={(_index, type) => isList(type) ? 96 : 160}
-          createCell={(type: string) => this.createCell(type)}
-          updateCell={(view, index) => (view as Cell).item = this._items[index].data}
-          onLastVisibleIndexChanged={this.handleLastVisibleIndexChanged}/>
+          onLastVisibleIndexChanged={({value})=> this.handleLastVisibleIndexChanged(value)}
+          onSelect={({item}) => this.onItemSelected.trigger({item: item as RedditPost})}>
+        <Cell
+            height={96}
+            selectable
+            itemCheck={() => isList(this.mode)}>
```

Comment:
This is kinda abusive (itemCheck is not inteded to be used like this), but so far I wasn't able to come up with a better idea. Maybe Cell needs a new feature to make this kind of behavior more convinient to implement.